lib/nadia/api.ex

```diff
@@ -40,14 +40,31 @@ defmodule Nadia.API do
   defp build_request(params, file_field) do
     params = params
     |> Keyword.update(:reply_markup, nil, &(Poison.encode!(&1)))
-    |> Enum.filter_map(fn {_, v} -> v end, fn {k, v} -> {k, to_string(v)} end)
+    |> Enum.filter_map(fn {_, v} -> v end, fn {k, v} -> {k, drop_nil_fields(v)} end)
     if !is_nil(file_field) and File.exists?(params[file_field]) do
       build_multipart_request(params, file_field)
     else
       {:form, params}
     end
   end
 
+  defp drop_nil_fields(value) when is_list(value) do
```

Comment:
for a nonanonymous function, maybe we can have a more meaningful name than `value`